subxt/src/rpc/lightclient/background.rs

```diff
@@ -0,0 +1,340 @@
+        BackgroundTask::default()
+    }
+
+    /// Handle the registration messages received from the user.
+    async fn handle_register(&mut self, message: BackendMessage) {
+        match message {
+            BackendMessage::Request { id, sender } => {
+                self.requests.insert(id, sender);
+            }
+            BackendMessage::Subscription { id, sender } => {
+                // Drain the subscription cache, that holds messages that
+                // were not propagated to this subscription yet (because the
+                // RPC server produced a notification before the user registered
+                // to receive notifications).
+                if let Some(cached_responses) = self.subscriptions_cache.remove(&id) {
+                    tracing::debug!(
+                        target: LOG_TARGET,
+                        "Some messages were cached before susbcribing"
+                    );
+
+                    for response in cached_responses {
+                        if sender.send(response).await.is_err() {
+                            tracing::warn!(
+                                target: LOG_TARGET,
+                                "Cannot send notification to subscription {:?}",
+                                id
+                            );
+                        }
+                    }
+                }
+
+                self.subscriptions.insert(id, sender);
+            }
+        };
+    }
+
+    /// Parse the response received from the light client and sent it to the appropriate user.
+    async fn handle_rpc_response(&mut self, response: String) {
+        match RpcResponse::from_str(&response) {
+            Ok(RpcResponse::Error { id, error }) => {
+                // Send the error back.
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender.send(error).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                }
+            }
+            Ok(RpcResponse::Method { id, result }) => {
+                // Send the response back.
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender.send(result).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                }
+            }
+            Ok(RpcResponse::Subscription { method, id, result }) => {
+                // Subxt calls into `author_submitAndWatchExtrinsic`, however the smoldot produces
+                // `{"event":"broadcasted","numPeers":1}` which is part of the RPC V2 API. Ignore
+                // this spurious event.
```

Comment:
Smoldot is also supposed to produce a `validated` event according to the new API. It doesn't because I haven't finished implementing it, but you might want to ignore it as well.